app/(public)/api/v2/links/[[...paths]]/route.ts

```diff
@@ -0,0 +1,75 @@
+import { ApiResponseHandler } from "@/lib/api-response";
+import { constants } from "@/lib/constants";
+import { createClient } from "@/lib/supaclient/server";
+import { NextRequest } from "next/server";
+
+const {
+  errors: {
+    link: { LINK_NOT_FOUND, LINK_MISSING_SIGNATURE, LINK_PATH_REQUIRED },
+    content: { CONTENT_NOT_FOUND },
+    general: { UNKNOWN },
+  },
+} = constants;
+
+export async function GET(
+  request: NextRequest,
+  { params }: { params: { paths: string[] } }
+) {
+  const searchParams = request.nextUrl.searchParams;
+  const supabase = createClient();
+  const linkPath = params.paths?.[0];
+  const appSignature = searchParams.get("app");
+
+  if (!appSignature) return ApiResponseHandler.error(LINK_MISSING_SIGNATURE);
+  if (appSignature !== process.env.NEXT_PUBLIC_APP_ID)
+    return ApiResponseHandler.error(LINK_MISSING_SIGNATURE);
+  if (!linkPath) return ApiResponseHandler.error(LINK_PATH_REQUIRED);
+
+  const { data: linkData, error: linkError } = await supabase
+    .from("link")
+    .select("*")
+    .eq("path", linkPath)
+    .limit(1)
+    .maybeSingle();
+
+  if (linkError) {
+    if (linkError.code === "PGRST116")
+      return ApiResponseHandler.error(LINK_NOT_FOUND);
+    return ApiResponseHandler.error(UNKNOWN);
+  }
+
+  if (!linkData) return ApiResponseHandler.error(LINK_NOT_FOUND);
+
+  const { data: contentData, error: contentError } = await supabase
+    .from("contents")
+    .select("*")
+    .eq("link_id", linkData.uuid)
+    .single();
+
+  if (contentError) {
+    if (contentError?.code === "PGRST116")
+      return ApiResponseHandler.error(CONTENT_NOT_FOUND);
+    return ApiResponseHandler.error(UNKNOWN);
+  }
```

Comment:
In this query, the preceding query on `link` is already guarded with `.limit(1)` and `.maybeSingle()`. Additionally, it includes a filter applied by its UUID. Given these safeguards, I believe it is very safe to assume this will always return either 1 result or none.